hosting/couchdb/Dockerfile

```diff
@@ -1,14 +1,105 @@
-FROM couchdb:3.2.1
+# Modified from https://github.com/apache/couchdb-docker/blob/main/3.2.1/Dockerfile
+FROM node:20-slim
+
+# Add CouchDB user account to make sure the IDs are assigned consistently
+RUN groupadd -g 5984 -r couchdb && useradd -u 5984 -d /opt/couchdb -g couchdb couchdb
```

Comment:
I'm wondering if there is someway we can pull the CouchDB image Dockerfile and switch its base dynamically, then use this as a local base image for our Couch image, would mean that we don't need to worry about keeping this base up to date - would just mean a funky `sed` to replace the from in the actual base image - could be worth a try to save some future pain?

Reply:
The amount of magic and hacks we're layering on top of each other to get things to work is worrying me. The `runner.sh` script already changes a lot about the configuration of this image depending on the environment it's running in. Dynamically pulling a Dockerfile and modifying it is a step too far for me, I think.

Reply:
> The amount of magic and hacks we're layering on top of each other to get things to work is worrying me. The `runner.sh` script already changes a lot about the configuration of this image depending on the environment it's running in. Dynamically pulling a Dockerfile and modifying it is a step too far for me, I think.

Hmm - my main concern is that we're depending an awful lot on there being no issues with the snapshot of the image we've used to create this - I agree there is a fair amount of magic, but it would be good to at least separate what is the Couch image from what is our image - giving us a little bit more control/knowledge of what is ours and what we would need to update in future.

Even if we were to split out the contents into its own Dockerfile, and add in some comments to explain where it is from that would help.

Reply:
Yeah - fair enough, this will have to do - we don't update this image too often anyway!